skore/src/skore/sklearn/train_test_split.py

```diff
@@ -0,0 +1,252 @@
+        Controls the shuffling applied to the data before applying the split. Pass an
+        int for reproducible output across multiple function calls.
+    shuffle : bool, default is True
+        Whether or not to shuffle the data before splitting. If shuffle=False
+        then stratify must be None.
+    stratify : array-like, optional
+        If not None, data is split in a stratified fashion, using this as the
+        class labels.
+    project : Project, optional
+        The project to save information into. If None, no information will be saved.
+
+    Returns
+    -------
+    splitting : list
+        List containing train-test split of inputs.
+        The length of the list is twice the number of arrays passed, including
+        the X and y keyword arguments. If arrays are passed positionally as well
+        as through X and y, the output arrays are ordered as follows: first the
+        arrays passed positionally, in the order they were passed, then X if it
+        was passed, then y if it was passed.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> X, y = np.arange(10).reshape((5, 2)), range(5)
+
+    # Drop-in replacement for sklearn train_test_split
+    >>> X_train, X_test, y_train, y_test = train_test_split(X, y,
+    ...     test_size=0.33, random_state=42)
+    >>> X_train
+    array([[4, 5],
+           [0, 1],
+           [6, 7]])
+
+    # Explicit X and y, makes detection of problems easier
+    >>> X_train, X_test, y_train, y_test = train_test_split(X=X, y=y,
+    ...     test_size=0.33, random_state=42)
+    >>> X_train
+    array([[4, 5],
+           [0, 1],
+           [6, 7]])
+
+    # When passing X and y explicitly, X is returned before y
+    >>> arr = np.arange(10).reshape((5, 2))
+    >>> arr_train, arr_test, X_train, X_test, y_train, y_test = train_test_split(
+    ...     arr, y=y, X=X, test_size=0.33, random_state=42)
+    >>> X_train
+    array([[4, 5],
+           [0, 1],
+           [6, 7]])
+    """
+    import sklearn.model_selection
+
+    new_arrays = list(arrays)
+    if X is not None:
+        new_arrays.append(X)
+    if y is not None:
+        new_arrays.append(y)
+
+    output = sklearn.model_selection.train_test_split(
+        *new_arrays,
+        test_size=test_size,
+        train_size=train_size,
+        random_state=random_state,
+        shuffle=shuffle,
+        stratify=stratify,
+    )
+
+    if y is None:
+        y = arrays[-1]
+
+    ml_task = _find_ml_task(y)
+
+    kwargs = dict(
+        arrays=new_arrays,
+        test_size=test_size,
+        train_size=train_size,
+        random_state=random_state,
+        shuffle=shuffle,
+        stratify=stratify,
+        y=y,
+        ml_task=ml_task,
+    )
+
+    for warning_class in [HighClassImbalanceWarning]:
+        check = warning_class.check(**kwargs)
+
+        if check is False:
+            warnings.warn(
+                message=warning_class.MSG, category=warning_class, stacklevel=1
+            )
```

Comment:
Probably in another task we could display those stings as formatted HTML if running in a notebook.

Maybe something like (not tested): 

```python
 with contextlib.suppress(ImportError):
        from IPython.core.interactiveshell import InteractiveShell
        from IPython.display import display

        if InteractiveShell.initialized():
            display("""
<div style="
  border: solid 1px var(--vscode-editorWarning-border, red);
  color: var(--vscode-editorWarning-foreground, white);
  background-color: var(--vscode-editorWarning-background, red);
">
  It seems that you have a classification problem with a high class 
  imbalance. In this 
  case, using train_test_split may not be a good idea because of high  
  variability in the scores obtained on the test set. 
  To tackle this challenge we suggest to use skore's 
  cross_validate function.
</div>""")
```

Reply:
+1 to control the way we show the warning in a notebook.

But, if i understand, you are proposing to print HTML content in a notebook, and raise a warning other-wise?
Isn't expose to side-effect if user can control its warnings (there is no more real warning in a notebook in your proposition, right?) ?

IMHO, a good trade-off would be to see how to customize the behavior of `warnings.warn` to our warnings in a notebook (i don't know if it's feasible).

Reply:
Globally yes, using `display` is also a way to enforce the visibility of our warnings as user may disable them.

Reply:
Globally yes, using display is also a way to enforce the visibility of our warnings as user may disable them.